src/dune_rules/dep_conf_eval.ml

```diff
@@ -27,30 +27,19 @@ let to_action_builder = function
   | Other x -> x
 
 let dep_on_alias_rec alias ~loc =
-  let src_dir = Path.Build.drop_build_context_exn (Alias.dir alias) in
-  Action_builder.of_memo (Source_tree.find_dir src_dir) >>= function
```

Comment:
Again, I think this combines general refactoring and adding a new feature which makes it hard to review. I would appreciate it if you did the refactoring separately (before or after the new feature is added).

Reply:
This is not a refactoring. We can't check whether the source tree exists here because it could be a melange-target-dir entrypoint that we're checking.